utils/config.go

```diff
@@ -250,8 +250,10 @@ func ReadConfig(r io.Reader, allowEnvironmentOverrides bool) (*model.Config, map
 	if unmarshalErr == nil {
 		// https://github.com/spf13/viper/issues/324
 		// https://github.com/spf13/viper/issues/348
-		config.PluginSettings = model.PluginSettings{}
-		unmarshalErr = v.UnmarshalKey("pluginsettings", &config.PluginSettings)
+		if _, ok := os.LookupEnv("MM_PLUGINSETTINGS"); ok {
```

Comment:
Maybe I'm missing something, but the linked issues talk about periods in the key names. Isn't the `UnmarshalKey` critical to ensure we parse the plugin settings data structure even if the plugin ids contain periods? And that we should be doing this independent of whether or not settings are coming from the environment?

Reply:
I think the purpose of the previous code was to explicitly call `v.UnmarshalKey` to set the entire PluginSettings object using `MM_PLUGINSETTINGS` because you can't use an environment variable like `MM_PLUGINSETTINGS_PLUGINS_COM.EXAMPLE.PLUGIN` because of the linked issues.